thousandeyes/util.go

```diff
@@ -311,14 +311,13 @@ func FixReadValues(m interface{}, name string) (interface{}, error) {
 		if e == nil && tp == nil {
 			m = nil
 		} else {
-			if e != nil {
-				m.(map[string]interface{})["email"] = e
-			}
-
+			// Add the third party map to the notifications map if they are present
+			// if they're not configured, then the API doesn't return them at all
 			if tp != nil {
 				m.(map[string]interface{})["third_party"] = tp
 			}
 
+			m.(map[string]interface{})["email"] = e
```

Comment:
so the issue here is that our api always needs the `email` attribute even if it doesn't have a value?

Reply:
Exactly! It always returns it, even if it's empty

Reply:
But the other issue here was that the notifications block is returned as a map by the API, but for terraform it's actually a Set (an unordered list), because maps have some limitations (you cannot force any specific keys unlike in Sets where you can force the type of elements). Before in terraform we were setting the entire email block to nil if the list of recipients was empty, but this was a problem in this logic, because then the email block stayed as a map, which was causing the panics. This lines ensures that the email block is always converted to a set, with or without recipients

Reply:
got it, thanks Pedro 😃